test/integration/model_validation_test.go

```diff
@@ -0,0 +1,251 @@
+				ObjectMeta: metadata("replicas-1-nil-2-valid"),
+				Spec: v1.ModelSpec{
+					URL:         "hf://test-repo/test-model",
+					Engine:      "VLLM",
+					Features:    []v1.ModelFeature{},
+					MinReplicas: 1,
+					Replicas:    nil,
+					MaxReplicas: ptr.To[int32](2),
+				},
+			},
+			expValid: true,
+		},
+		{
+			model: v1.Model{
+				ObjectMeta: metadata("replicas-1-2-nil-valid"),
+				Spec: v1.ModelSpec{
+					URL:         "hf://test-repo/test-model",
+					Engine:      "VLLM",
+					Features:    []v1.ModelFeature{},
+					MinReplicas: 1,
+					Replicas:    ptr.To[int32](2),
+					MaxReplicas: nil,
+				},
+			},
+			expValid: true,
+		},
+		{
+			model: v1.Model{
+				ObjectMeta: metadata("replicas-3-nil-2-invalid"),
+				Spec: v1.ModelSpec{
+					URL:         "hf://test-repo/test-model",
+					Engine:      "VLLM",
+					Features:    []v1.ModelFeature{},
+					MinReplicas: 3,
+					MaxReplicas: ptr.To[int32](2),
+				},
+			},
+			expErrContain: "minReplicas should be less than or equal to maxReplicas",
+		},
+		{
+			model: v1.Model{
+				ObjectMeta: metadata("cache-profile-with-hf-url-valid"),
+				Spec: v1.ModelSpec{
+					URL:          "hf://test-repo/test-model",
+					Engine:       "VLLM",
+					Features:     []v1.ModelFeature{},
+					CacheProfile: "some-cache-profile",
+				},
+			},
+			expValid: true,
+		},
+		{
+			model: v1.Model{
+				ObjectMeta: metadata("cache-profile-with-non-hf-url-invalid"),
+				Spec: v1.ModelSpec{
+					URL:          "ollama://test-repo/test-model",
+					Engine:       "VLLM",
+					Features:     []v1.ModelFeature{},
+					CacheProfile: "some-cache-profile",
+				},
+			},
+			expErrContains: []string{
+				"cacheProfile is only supported with a huggingface url",
+				"hf://",
+			},
+		},
+		{
+			model: v1.Model{
+				ObjectMeta: metadata("update-no-changes-valid"),
+				Spec: v1.ModelSpec{
+					URL:      "hf://test-repo/test-model",
+					Engine:   "VLLM",
+					Features: []v1.ModelFeature{},
+				},
+			},
+			update:   func(m *v1.Model) { /* No changes */ },
+			expValid: true,
+		},
+		{
+			model: v1.Model{
+				ObjectMeta: metadata("mutate-url-invalid"),
+				Spec: v1.ModelSpec{
+					URL:      "hf://test-repo/test-model",
+					Engine:   "VLLM",
+					Features: []v1.ModelFeature{},
+				},
+			},
+			update: func(m *v1.Model) {
+				m.Spec.URL = "hf://update-test-repo/update-test-model"
+			},
+			expErrContain: "url is immutable",
+		},
+		{
+			model: v1.Model{
+				ObjectMeta: metadata("mutate-cacheprofile-invalid"),
+				Spec: v1.ModelSpec{
+					URL:          "hf://test-repo/test-model",
+					Engine:       "VLLM",
+					Features:     []v1.ModelFeature{},
+					CacheProfile: "some-cache-profile",
+				},
+			},
+			update: func(m *v1.Model) {
+				m.Spec.CacheProfile = "some-updated-cache-profile"
+			},
+			expErrContain: "cacheProfile is immutable",
+		},
+	}
+	for i := range cases {
+		// Copy case to avoid parallel access issues with the use of range.
+		c := cases[i]
```

Comment:
nit: the sharing "issue" was solve in [Go 1.22](https://tip.golang.org/doc/go1.22) .  you can simply use `for _, c  := range cases {`

Reply:
Filed #277 to track to avoid scope creep

Reply:
Good to know!